### PR TITLE
Rework user cancellation workflow to do immediate cancellation

### LINKED ISSUE #775: Self-cancellation configuration and test
- [x] Configure the user cancellation confirmation email
- [x] Remove the confirmation email step, because it would require them to log in (#779) - See [stackexchange](http://drupal.stackexchange.com/a/32427/449)
- [ ] Specify new items for translators
- [x] Links like user/UID/delete become user/UID/cancel (See admin emails especially). Consider adding a URL alias, consider leaving the name of the button "Delete account"
- [x] Verify admin workflow - should be no confirmation email required for spammers, but yes for requests by members?
- [x] Review user workflow per #779 - Should we even do the email confirmation thing?

### PR DESCRIPTION
**Pull request by rfay**
This PR fixes #775 by 
- adding the [user_cancel_immediate](drupal.org/project/user_cancel_immediate) module, which I just posted on drupal.org - it was written for this issue.
- Adding a bit more to the warning text, and making the "cancel" link more understandable.

### CODE REVIEW

File: docroot/sites/all/modules/contrib/user_cancel_immediate/user_cancel_immediate.module [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+ * sending email and having them click the link.
+ * Suggested by http://drupal.stackexchange.com/a/32427/449
+ *
+ * @param $form
+ * @param $form_state
+ */
+function _user_cancel_immediate_user_cancel_form_submit(&$form, &$form_state) {
+  $uid = $form['uid']['#value'];
+
+  if (!user_access('administer users') && $uid == $GLOBALS['user']->uid) {
```
Comment (kalpaitch) [inline docroot/sites/all/modules/contrib/user_cancel_immediate/user_cancel_immediate.module]: Would this not run into trouble if the 'user cancel own account' permission is enabled for ordinary users?

File: docroot/sites/all/modules/contrib/user_cancel_immediate/user_cancel_immediate.module [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+ * Thanks to Clive, http://drupal.stackexchange.com/a/32427/449
+ */
+
+
+/**
+ * Implements hook_form_FORMID_alter()
+ * Adds a custom submit handler
+ */
+function user_cancel_immediate_form_user_cancel_confirm_form_alter(&$form, &$form_state, $form_id) {
+  $form['#submit'][] = '_user_cancel_immediate_user_cancel_form_submit';
```
Comment (kalpaitch) [inline docroot/sites/all/modules/contrib/user_cancel_immediate/user_cancel_immediate.module]: Don't see why we can't just alter the 'user_cancel_confirm' form element to either permanently disable the confirmation email option or to allow a choice for all users:

$form['user_cancel_confirm'] = array(
    '#type' => 'checkbox',
    '#title' => t('Require e-mail confirmation to cancel account.'),
    '#default_value' => FALSE,
    '#access' => FALSE,
    '#description' => t('When enabled, the user must confirm the account cancellation via e-mail.'),
  );

Comment (kalpaitch) [inline docroot/sites/all/modules/contrib/user_cancel_immediate/user_cancel_immediate.module]: Ignore that last part, I see user_cancel_confirm_form_submit assumes (incorrectly) that if the value is false the user must be an admin.

**Comment by kalpaitch** [comment]
@rfay Dirty but look OK. I'd prefer a user_cancel_delay type module if we're going to write something custom and have cron cleanup after three days like we discussed, but this is a bit more work and not neccessary now.

**Comment by rfay** [comment]
I definitely agree - I just don't want to tie that to the upgrade. I love the idea and wrote it up in #823 - See if I got the gist there.

**Pull request merged**